Neos.Flow/Classes/Mvc/Exception/StopActionException.php

```diff
@@ -23,7 +23,24 @@
  *
  * @api
  */
-class StopActionException extends \Neos\Flow\Mvc\Exception
+final class StopActionException extends \Neos\Flow\Mvc\Exception
 {
-    public ActionResponse $response;
+    /**
+     * As throwing the exception allows for an unusual control flow, we attach the response for the dispatcher.
+     */
+    public readonly ActionResponse $response;
+
+    private function __construct(string $message, int $code, ?\Throwable $previous, ActionResponse $response)
+    {
+        parent::__construct($message, $code, $previous);
+        $this->response = $response;
+    }
+
+    public static function create(
+        ActionResponse $response,
+        string $message,
+        int $code
```

Comment:
im unsure _why_ there even has to be a message and a code?

Isnt this purely for control flow?

Reply:
nod, but then that's what exceptions are for? :D

Reply:
it _could_ extract the message from the request on its own.

```php
$infoMessage = sprintf(
    '%s %s',
    $response->getStatusCode(),
    ResponseInformationHelper::getStatusMessageByCode($response->getStatusCode())
);
```

But then again we are used to specifying a message (wich will be used for the response content as well)

https://github.com/neos/flow-development-collection/blob/a5cd6e34e777a0e3658d5ef4bb9c1528381fcb04/Neos.Flow/Classes/Mvc/Controller/RestController.php#L68

So lets keep the `$infoExceptionMessage` only for informational purposes, but make it optional.
I would fix the exception timestamp code to a fix number though.

Reply:
Fine